src/Features/Core/Portable/SolutionCrawler/WorkCoordinator.SemanticChangeProcessor.cs

```diff
@@ -339,6 +353,20 @@ private class ProjectProcessor : IdleProcessor
                         _pendingWork = new Dictionary<ProjectId, Data>();
 
                         Start();
+
+                        AsyncProcessorTask.ContinueWith(
+                            _ =>
+                            {
+                                foreach (var (projectId, data) in _pendingWork)
+                                {
+                                    data.AsyncToken.Dispose();
+                                }
+
+                                _pendingWork.Clear();
+                            },
+                            CancellationToken.None,
+                            TaskContinuationOptions.ExecuteSynchronously,
+                            TaskScheduler.Default);
```

Comment:
duplicate code.  Can we we share with above?

Reply:
Should we be using LazyCancellation here?  in other words, is it ok if this code runs before some antecedent tasks completes, but after the cancellation token controlling another intermediary tasks completes?  In other words, should this be using the semantics of SafeContinueWith?

Reply:
Doesn't `LazyCancellation` only have an effect on continuations that specify a `CancellationToken`?

Reply:
I don't think so (but i'm not totally sure). I thought it was more about the *antecedent* tasks and if *they* used a cancellation token.  in other words, if i have tasks:

```A <- B <- C```   (i.e. B depends on A and C depends on B)

And 'A' is running, but B gets canceled.  The question is: should 'C' run?  LazyCancellation (to me) says: "No.  All antecedent tasks are not completed.   Even though B transitioned ot Canceled, C should not run yet because 'A' is still running".  

This is regardless of the cancellation token passed to C.

Reply:
@stephentoub For guidance.

Reply:
💭 I looked at referencesource.microsoft.com and it *seems* to only impact cases where a cancellation token is involved.

Reply:
@CyrusNajmabadi is correct re "I thought it was more about the antecedent tasks and if they used a cancellation token." If you have:
```C#
Task a = ...;
Task b = a.ContinueWith(..., token);
Task c = b.ContinueWith(...);
```
and while a is running the token has cancellation requested, should c be allowed to run immediately or must it wait for a to finish.

Reply:
@stephentoub I believe you may have misunderstood the context. In my code, I pass `CancellationToken.None` to `ContinueWith`. I am claiming that `TaskContinuationOptions.LazyCancellation` will never have an impact on a call to `ContinueWith` called in this manner.

Reply:
I didn't look at your code; I was commenting on the behavior of LazyCancellation and what it actually does.

Given the code:
```C#
Task a = ...;
Task b = a.ContinueWith(..., token);
Task c = b.ContinueWith(...);
```
`c` can end up running concurrently with `a`.  To prevent that, you can make it:
```C#
Task a = ...;
Task b = a.ContinueWith(..., token, TaskContinuationOptions.LazyCancellation, TaskScheduler.Default);
Task c = b.ContinueWith(...);
```
and then `c` will not run concurrently with `a`.  If that `token` is `None`, then it doesn't matter because it won't be canceled prior to `a` completing (it could still be canceled by other means, e.g. if you passed TaskContinuationOptions.OnlyOnFaulted and the antecedent completed in another state, but in such cases the antecedent needs to be completed for that cancellation to happen).

Reply:
If this is the case, can we comment that it's ok for us to not pass LazyCancellation here as it seems to only be relevant for the ContinueWith's that take cancellation tokens?

Thanks!